backend/src/main/java/com/myshop/products/Product.java

```diff
@@ -1,126 +1,106 @@
 package com.myshop.products;
 
-import com.myshop.utils.SLUGIFY;
-
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonSetter;
+import com.myshop.categories.Category;
+import lombok.Getter;
+import lombok.Setter;
 import javax.persistence.*;
-import javax.validation.constraints.NotBlank;
 import javax.validation.constraints.NotNull;
+import java.sql.Date;
 import java.util.List;
 
 @Entity
-@Table( name = "products",
-        indexes =@Index(columnList = "name_id", unique = true))
-
+@Setter
+@Getter
+@Table(name = "products",
+        indexes = {
+                @Index(columnList = "id", unique = true),
+                @Index(columnList = "name")
+        })
 public class Product {
     @Id
     @GeneratedValue(strategy = GenerationType.SEQUENCE)
     @Column(name = "id", nullable = false)
     private Long id;
-    @Column(name = "name_id", nullable = false)
-    @NotBlank(message = "Name is required")
-    private String nameId;
-    @NotNull
+    @NotNull(message = "Name is required")
     private String name;
-    @NotNull
+    @NotNull(message = "Price is required")
     private float price;
-    @NotNull
+    @NotNull(message = "ImageURL is required")
     private String imageUrl;
     @ElementCollection
     private List<String> color;
     @ElementCollection
     private List<String> size;
     private String description;
+    @ManyToOne(fetch = FetchType.EAGER)
```

Comment:
please help me know more about fetch types and what are the diff between them.

Reply:
-LAZY: when execute a find or select query, result **not** return with relevant Category. So we need to execute an addition query to get Category. However, performance of LAZY is better
- EAGER: when execute a find or select query, result will include relevant Category. Sometimes, those categories are redundant and unused.